src/EnumBitMask.php

```diff
@@ -19,16 +21,27 @@ final class EnumBitMask implements BitMaskInterface
 
     /**
      * @param class-string $enum
+     * @throws InvalidEnumException
      * @throws UnknownEnumException
      */
     public function __construct(
         private readonly string $enum,
         int $mask = 0,
+        bool $isIntBacked = false,
     ) {
         if (!is_subclass_of($this->enum, UnitEnum::class)) {
             throw new UnknownEnumException('EnumBitMask enum must be subclass of UnitEnum');
         }
         foreach ($this->enum::cases() as $index => $case) {
+            if ($isIntBacked) {
+                if (!is_subclass_of($this->enum, BackedEnum::class)) {
+                    throw new InvalidEnumException('Enum must be a backed enum');
+                }
+                /** @phpstan-var BackedEnum $case */
+                if (!is_int($case->value)) {
+                    throw new InvalidEnumException('Enum must be an int-backed enum with integer values');
+                }
+            }
             $this->map[$case->name] = Bits::indexToBit($index);
         }
         $this->bitmask = new BitMask($mask, count($this->enum::cases()) - 1);
```

Comment:
Not sure what the difference is, but I had an issue here with the most significant bit in my implementation. Not sure exactly why, but removing the second argument solved the issue for me. Perhaps you can speculate what could be the cause this?

Reply:
Regarding the issue with the most significant bit, it’s possible that the problem stems from how the second argument (the mask) interacts with the bit positions in your enum. If the bitmask value overlaps or conflicts with the most significant bit, removing the second argument might have bypassed the issue. I’ll investigate further to pinpoint the exact cause.